src/esi.cls.php

```diff
@@ -771,6 +779,8 @@ public function sub_admin_bar_block()
 		);
 
 		echo $this->sub_esi_block('admin-bar', 'adminbar', $params);
+		
+		$this->admin_rendered = true;
```

Comment:
This line is better moved to above line 781. That can give an additional safe check to avoid infinite embed just in case in `$this->sub_esi_block()` it called some hook which fall into this func again. How do you think?